### ISSUE TITLE
rustc shouldn't say 'not in scope' when using a derive macro as a macro_rules macro

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#76429]
Consider the following code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=81691f8ee3b6f001133682563604ced5)):
```rust
fn f() {
    Debug!()
}

#[derive(Debug)]
struct S;
```
```
error: cannot find macro `Debug` in this scope
 --> src/lib.rs:2:5
  |
2 |     Debug!()
  |     ^^^^^
```
This is not particularly helpful (in fact it's incorrect). `Debug` _is_ in scope, it's just not a _macro rules_ macro. Instead the error should say something like
```rust
error: `Debug` is a proc_macro, not a macro_rules! macro
 --> src/lib.rs:2:5
  |
2 |     Debug!()
  |     ^^^^^ note: attempting to use a proc_macro as a macro_rules macro
```

**Comment by jyn514** [comment]
This would have caught the second 'library error' in https://jyn514.github.io/2020/09/05/Rust-in-2021.html.

**Comment by petrochenkov** [comment]
The desirable error was reported previously, but it changed with introduction of two sub-namespaces in macro namespace - https://github.com/rust-lang/rust/pull/54069.

So, derive `Debug` is never found during regular name lookup now for names in fn-like macro positions (`Debug!`).
So it needs a second lookup for diagnostics, which is currently done for other namespaces, but not other sub-namespaces in the macro namespace.
Compare with:
```rust
type A = Debug; // ERROR: expected type, found derive macro `Debug`
```

**Comment by dario23** [comment]
Current playground correctly says
```
error: cannot find macro `Debug` in this scope
 --> src/lib.rs:2:5
  |
2 |     Debug!()
  |     ^^^^^
  |
  = note: `Debug` is in scope, but it is a derive macro: `#[derive(Debug)]`
```
so let's close this? @jyn514

**Comment by jyn514** [comment]
oh amazing, good find!
